changelog/unreleased/breaking-changes/2829--change-reader-and-writer-plugin-api.md

```diff
@@ -0,0 +1,4 @@
+The reader_options method for a reader plugin and writer_options method for a
+writer plugin must return 'vast::config_options'. The new type allows for a
+customization in the CLI arguments parsing logic that may be needed in the
+future without breaking the API as it is owned by VAST.
```

Comment:
I don't think this is user-facing, but rather developer-facing. No changelog entry should be necessary.